pkg/build/controller/build/build_controller.go

```diff
@@ -350,6 +351,16 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 		}
 	}
 
+	// set the pushSecret that will be needed by the build to push the image to the registry
+	// at the end of the build.
+	build.Spec.Output.PushSecret = secret
+
+	// ensure the build object the pod sees starts with a clean set of reasons/messages,
+	// rather than inheriting the potential "invalidoutputreference" message from the current
+	// build state.
+	build.Status.Reason = ""
```

Comment:
yes, it's picked up by the code running inside the build pod when that code goes to update the build object w/ the source revision.  That code starts from the serialized build object (what we're updating here) and then adds the revision and uses it to post an update to the build object.  If we don't clear this value here, the update will include the stale build.Status.Reason.